moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkExecutor.h

```diff
@@ -0,0 +1,207 @@
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Rice University nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Ryan Luna */
+
+#ifndef MOVEIT_ROS_BENCHMARKS_BENCHMARK_EXECUTOR_
+#define MOVEIT_ROS_BENCHMARKS_BENCHMARK_EXECUTOR_
+
+#ifdef USE_BOOST_FILESYSTEM_V2
+#define BOOST_FILESYSTEM_VERSION 2
+#endif
+
+#include "moveit/benchmarks/BenchmarkOptions.h"
+
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+
+#include <moveit/warehouse/planning_scene_storage.h>
+#include <moveit/warehouse/planning_scene_world_storage.h>
+#include <moveit/warehouse/state_storage.h>
+#include <moveit/warehouse/constraints_storage.h>
+#include <moveit/warehouse/trajectory_constraints_storage.h>
+#include <moveit/planning_interface/planning_interface.h>
+#include <warehouse_ros/database_loader.h>
+#include <pluginlib/class_loader.h>
+
+#include <map>
+#include <vector>
+#include <string>
+#include <boost/function.hpp>
+
+namespace moveit_ros_benchmarks
+{
+
+/// A class that executes motion plan requests and aggregates data across multiple runs
+/// Note: This class operates outside of MoveGroup and does NOT use PlanningRequestAdapters
+class BenchmarkExecutor
+{
+public:
+    /// Structure to hold information for a single run of a planner
+    typedef std::map<std::string, std::string> PlannerRunData;
+    /// Structure to hold information for a single planner's benchmark data.
+    typedef std::vector<PlannerRunData> PlannerBenchmarkData;
+
+    /// Definition of a query-start benchmark event function.  Invoked before a new query is benchmarked.
+    typedef boost::function<void(const moveit_msgs::MotionPlanRequest& request, planning_scene::PlanningScenePtr)> QueryStartEventFunction;
```

Comment:
since we are not backporting this, I think it would be better to go ahead and switch all the boost references to std where possible